### PR TITLE
[ECR] Fix project secret overriding docker registry secret

### PR DESCRIPTION
**Pull request by alonmr**
https://iguazio.atlassian.net/browse/ML-5960

We already did this fix for IAM policy in https://github.com/mlrun/mlrun/pull/3407 but we didn't realize it is also needed for not overriding docker registry secret.

### CODE REVIEW

File: server/api/utils/builder.py [hunk 1]
```diff
@@ -340,6 +327,17 @@ def configure_kaniko_ecr_init_container(
...
+    # project secret might conflict with the attached instance role/docker registry secret
+    # ensure "AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY" have no values or else kaniko will fail
+    # due to credentials conflict / lack of permission on given credentials
+    kpod.env = list(
+        filter(
+            lambda env_var: env_var.name
+            not in ["AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY"],
+            kpod.env,
+        )
+    )
```
Comment (moranbental) [inline server/api/utils/builder.py]:
```suggestion
kpod.env = [
    env_var for env_var in kpod.env
    if env_var.name not in ["AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY"]
]
```

File: tests/api/utils/test_builder.py [hunk 2]
```diff
@@ -492,17 +492,14 @@ def test_build_runtime_ecr_with_ec2_iam_policy(monkeypatch):
         env.to_dict() for env in pod_spec.containers[0].env
     ]
 
-    # ensure both envvars are set without values, so they won't interfere with the iam policy
+    # ensure both envvars are not set, so they won't interfere with the iam policy/docker registry secret
     for env_name in ["AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY"]:
-        assert {"name": env_name, "value": "", "value_from": None} in [
-            env.to_dict() for env in pod_spec.containers[0].env
-        ]
+        assert env_name not in [env.name for env in pod_spec.containers[0].env]
```
Comment (moranbental) [inline tests/api/utils/test_builder.py]: no need to create a list 
```suggestion
        assert env_name not in (env.name for env in pod_spec.containers[0].env)
```

**Review by moranbental — CHANGES_REQUESTED** [review]
Just 2 comments bellow

**Review by moranbental — APPROVED** [review]
LGTM

**Review by liranbg — APPROVED** [review]

**Pull request merged**